src/quic/frame.d

```diff
@@ -0,0 +1,86 @@
+module quic.frame;
+import quic.attributes;
+
+enum tlsFrameTypes {
+    ClientHello, ServerHello
+}
+
+enum tlsExtensionTypes {
+   SupportedGroups = 10, SupportedVersions = 43, KeyShare = 51 
+}
+
+alias VarInt = ulong;
+
+struct AckFrame {
+    ubyte frameType;
+    VarInt largestAcknowledged;
+    VarInt ackDelay;
+    AckRange[] range;
+    EcnCount[] ecnCounts;
+    @VarIntLength ubyte[] ackRanges;
+}
+
+struct AckRange {
+    VarInt gap;
+    VarInt rangeLength;
+}
+
+struct EcnCount {
+    VarInt ect0Count;
+    VarInt ect1Count;
+    VarInt ecn_ceCount;
+}
+
+struct CryptoFrame
+{
+    VarInt type;
+    VarInt offset;
+    VarInt length;
+    @VarIntLength ubyte[] cryptoData;
+}
+
+//TLS 1.3 frames
+//Reference : RFC 8446
+
+alias TlsData = ubyte[2];
+
+@TlsFrame!(tlsFrameTypes.ClientHello) @FixedLength!3 struct ClientHello
+{
+    uint legacy_version;
+    @EstablishedLength!32 ubyte[32] random;
+    //not used by QUIC
+    ubyte[] legacy_session_id = [0];
+    //not used by QUIC
+    @FixedLength!1 ubyte[] legacy_compression_method = [0];
+    @FixedLength!2 ubyte[] cipher_suites = [0x13, 0x1];//TLS_AES_128_GCM_SHA256
```

Comment:
you might want to be careful with these assignments. it will refer to a single shared static array, meaning if someone ever writes to cipher_suites[0], it will affect all the ClientHello things! You might consider making it immutable.